File: core/src/main/java/de/qabel/core/repository/ClientConfigRepository.java
package de.qabel.core.repository;

import de.qabel.core.repository.exception.EntityNotFoundException;
import de.qabel.core.repository.exception.PersistenceException;

public interface ClientConfigRepository {
    String find(String key) throws EntityNotFoundException, PersistenceException;
    boolean contains(String key) throws PersistenceException;
    void save(String key, String value) throws PersistenceException;
}

File: core/src/main/java/de/qabel/core/repository/ClientConfigRepository.kt
package de.qabel.core.repository

import de.qabel.core.repository.exception.EntityNotFoundException
import de.qabel.core.repository.exception.PersistenceException

interface ClientConfigRepository {
    @Throws(EntityNotFoundException::class, PersistenceException::class)
    fun find(key: String): String

    @Throws(PersistenceException::class)
    operator fun contains(key: String): Boolean

    @Throws(PersistenceException::class)
    fun save(key: String, value: String)
}
